devito/operations/interpolators.py

```diff
@@ -388,3 +389,520 @@ def callback():
+
+        # Values ​​that define which cubic kernel will be used
+        option = [2, 1, 1, 2]
+
+        pos = sympy.Matrix(position)
+
+        dim_pos = sympy.Matrix(dim_pos)
+
+        # Neighboring points(interpolation) or source value (injection)
+        p = [expr.xreplace(subs) for subs in idx_subs]
+        p = sympy.Matrix(p)
+
+        # Defines the kernel witch will be multiplied by the neighboring
+        # points(or source value)
+        base_index = [idx3d, idx2d]
+        index_opt = [b for b in base_index if b is not None]
+        arg_kernel = [pp - dim for pp, dim in zip(pos, dim_pos)]
+
+        kernel = [cubicConv(arg, option[opt]) for arg, opt in zip(arg_kernel, index_opt)]
+        kernel = np.prod(kernel)
+        kernel = [kernel*cubicConv(arg_kernel[-1], option[ii]) for ii in range(4)]
+
+        # Defining the final shape of the cubic equation
+        points = [point*k.subs(idx) for point, k, idx in zip(p, kernel, idx_subs)]
+
+        return points
+
+    def _bicubic_equations(self, expr, idx_subs, dim_pos=None):
+        """
+        Generate equations responsible for bicubic interpolation's computation.
+        Parameters
+        ----------
+        expr : expr-like
+            Input expression to interpolate.
+        idx_subs: dict
+            Structure responsible for mapping the order of substitution
+            of dimensions in expr.
+        dim_pos: tuple
+            Structure containing the spatial dimensions of the field
+        """
+        pos = self.sfunction._point_symbols
+
+        eqs = [self._cubic_equation(expr, pos, dim_pos,
+               idx_subs=idx_subs[ii*4:(ii+1)*4], idx2d=ii)
+               for ii in range(4)]
+        return eqs
+
+    def _tricubic_equations(self, expr, idx_subs, dim_pos=None):
+        """
+        Generate equations responsible for tricubic interpolation's computation.
+        Parameters
+        ----------
+        expr : expr-like
+            Input expression to interpolate.
+        idx_subs: dict
+            Structure responsible for mapping the order of substitution
+            of dimensions in expr.
+        dim_pos: tuple
+            Structure containing the spatial dimensions of the field
+        """
+        pos = self.sfunction._point_symbols
+
+        eqs = []
+        eqs.extend([self._cubic_equation(expr, pos, dim_pos,
+                    idx_subs=idx_subs[(ind*4)+(16*ii):((1+ind)*4)+16*ii],
+                    idx2d=ind, idx3d=ii)
+                    for ii in range(4) for ind in range(4)])
+
+        return eqs
+
+    def _interpolation_indices(self, variables, offset=0, field_offset=0):
+        """
+        Generate interpolation indices for the DiscreteFunctions in ``variables``.
+        """
+        index_matrix, points = self.sfunction._index_matrix(offset)
+
+        idx_subs = []
+        for i, idx in enumerate(index_matrix):
+            # Introduce ConditionalDimension so that we don't go OOB
```

Comment:
This should only be necessary if `r > variables.space_order`

Reply:
So if  `r <= variables.space_order`  there is no need to use  ConditionalDimension in this case?

Reply:
Yes

Reply:
I updated the PR with this fix

Reply:
I adjust this part of the code and updated here. 

You'll see that I added a conditional`if hasattr(x, 'space_order') else False`. I needed to do this because in some cases `variables` does not have the `space_order` attribute